src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -491,7 +537,16 @@ public Money GetAddressBalance(string address, int minConfirmations = 1)
         public Money GetReceivedByAddress(string address, int minConfirmations = 1)
         {
             if (this.addressIndexRepository == null)
+            {
+                this.logger.LogTrace("(-)[NOT_INITIALIZED]");
                 throw new IndexerNotInitializedException();
+            }
+
+            if (!this.IsSynced())
+            {
+                this.logger.LogTrace("(-)[NOT_SYNCED]");
+                throw new OutOfSyncException();
```

Comment:
I don't think that it is right that we throw exceptions like this. Exceptions are just that, exceptions. They should be seen as exceptional circumstances where something very bad has gone wrong in the code. To me this is a very valid scenario to check and thus we would just rather return some class that holds the result of `GetReceivedByAddress` along with a message or flag indicating that the call failed or not.